dandi/utils.py

```diff
@@ -618,14 +618,27 @@ def get_instance(dandi_instance_id):
         raise CliVersionTooOldError(our_version, minversion, bad_versions)
     if our_version in bad_versions:
         raise BadCliVersionError(our_version, minversion, bad_versions)
-    services = server_info["services"]
-    return dandi_instance(
-        metadata_version=0,
-        girder=services.get("girder", {}).get("url"),
-        gui=services.get("webui", {}).get("url"),
-        redirector=redirector_url,
-        api=services.get("api", {}).get("url"),
-    )
+    if server_info["services"].get("girder"):
+        return dandi_instance(
+            metadata_version=0,
+            girder=server_info["services"]["girder"]["url"],
+            gui=server_info["services"]["webui"]["url"],
+            redirector=redirector_url,
+            api=None,
+        )
+    elif server_info["services"].get("api"):
+        return dandi_instance(
+            metadata_version=1,
+            girder=None,
+            gui=server_info["services"]["webui"]["url"],
+            redirector=redirector_url,
+            api=server_info["services"]["api"]["url"],
+        )
```

Comment:
could we please retain robustness of `.get` to provide `None`? or what would be the benefit of client crashing here if redirector again just drops a field instead of explicitly annotating with `null`?

Reply:
Done.

Reply:
Thank you! next one to touch this piece of code is to reduce duplication.  Let's proceed